protocol/communities/manager.go

```diff
@@ -961,6 +1043,512 @@ func (m *Manager) GetAdminCommunitiesChatIDs() (map[string]bool, error) {
+		}
+		if lastArchiveEndDateTimestamp == 0 {
+			// This means there's no waku message stored for this community so far
+			// (even after requesting possibly missed messages), so no messages exist yet that can be archived
+			return 0, nil
+		}
+	}
+
+	return lastArchiveEndDateTimestamp, nil
+}
+
+func (m *Manager) CreateAndSeedHistoryArchive(communityID types.HexBytes, topics []types.TopicType, startDate time.Time, endDate time.Time, partition time.Duration) error {
+	m.UnseedHistoryArchiveTorrent(communityID)
+	err := m.CreateHistoryArchiveTorrent(communityID, topics, startDate, endDate, partition)
+	if err != nil {
+		return err
+	}
+	return m.SeedHistoryArchiveTorrent(communityID)
+}
+
+func (m *Manager) StartHistoryArchiveTasksInterval(community *Community, interval time.Duration, dispatchMagnetlink func(types.HexBytes)) {
+	id := community.IDString()
+	_, exists := m.historyArchiveTasks[id]
+
+	if exists {
+		log.Println("History archive tasks interval already runs for community: ", id)
+		m.logger.Debug("History archive tasks interval already runs for community: ", zap.Any("id", id))
+		return
+	}
+
+	cancel := make(chan struct{})
+	m.historyArchiveTasks[id] = cancel
+	m.historyArchiveTasksWaitGroup.Add(1)
+
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	log.Println("Starting history archive tasks interval for community: ", id)
+	m.logger.Debug("Starting history archive tasks interval for community: ", zap.Any("id", id))
+	for {
+		select {
+		case <-ticker.C:
+			log.Println("Executing history archive tasks for community: ", id)
+			lastArchiveEndDateTimestamp, err := m.GetHistoryArchivePartitionStartTimestamp(community.ID())
+			if err != nil {
+				m.logger.Debug("failed to get last archive end date", zap.Error(err))
+				continue
+			}
+
+			if lastArchiveEndDateTimestamp == 0 {
+				// This means there are no waku messages for this community,
+				// so nothing to do here
+				continue
+			}
+
+			topics, err := m.GetCommunityChatsTopics(community.ID())
+			if err != nil {
+				m.logger.Debug("failed to get community chats topics", zap.Error(err))
+				continue
+			}
+
+			to := time.Unix(time.Now().Unix(), 0)
+			lastArchiveEndDate := time.Unix(int64(lastArchiveEndDateTimestamp), 0)
+
+			err = m.CreateAndSeedHistoryArchive(community.ID(), topics, lastArchiveEndDate, to, interval)
+			if err != nil {
+				m.logger.Debug("failed to create and seed history archive", zap.Error(err))
+				continue
+			}
+			dispatchMagnetlink(community.ID())
+		case <-cancel:
+			m.UnseedHistoryArchiveTorrent(community.ID())
+			delete(m.historyArchiveTasks, id)
+			m.historyArchiveTasksWaitGroup.Done()
+			return
+		}
+	}
+}
+
+func (m *Manager) StopHistoryArchiveTasksIntervals() {
+	for _, t := range m.historyArchiveTasks {
+		close(t)
+	}
+	// Stoping archive interval tasks is async, so we need
+	// to wait for all of them to be closed before we shutdown
+	// the torrent client
+	m.historyArchiveTasksWaitGroup.Wait()
```

Comment:
@Samyoul And wait for it to be done here, making `StopHistoryArchiveTasksIntervals` a somewhat synchronous API